StantUiIosLibrary/Classes/Utils/UIImageViewExtension.swift

```diff
@@ -9,19 +9,31 @@ import UIKit
 
 extension UIImageView {
     
-    func showRounded(image: String, radius: CGFloat = 8) {
+    func showRounded(image: String, radius: CGFloat = 8, activityIndicator: UIActivityIndicatorView = UIActivityIndicatorView()) {
         self.layer.cornerRadius  = radius
         self.layer.masksToBounds = true
         
-        self.show(image: image)
+        self.show(image: image, activityIndicator: activityIndicator)
     }
     
-    func show(image: String) {
-        if let url = NSURL(string: image) {
-            if let data = NSData(contentsOf: url as URL) {
-                self.image = UIImage(data: data as Data)
+    func show(image: String, activityIndicator: UIActivityIndicatorView) {
+        activityIndicator.startAnimating()
+        DispatchQueue.global(qos: .background).async {
+            var imageViewImage = UIColor.stantImagePlaceholder.image()
+            if !image.isEmpty {
```

Comment:
It could be helpful to try to avoid a lot of if's. Alternatively,  we can extract the creation of UIImage to another method or even an extension of UIImage (we are using this conversion in many places). What do you think?

Reply:
In this case I don't think that is a good approach the extension because we are not changing proprieties of a uiimage itself. But I'm refactoring code to make it more legible :)